pkg/snapshot/rule/rule.go

```diff
@@ -251,19 +258,35 @@ func executeSnapRule(pvcs []v1.PersistentVolumeClaim, snap *crdv1.VolumeSnapshot
 					}
 
 					for _, action := range item.Actions {
+						if action.Background {
+							backgroundActionPresent = true
+						}
+
 						if action.Type == apis_stork.StorkRuleActionCommand {
-							err := executeCommandAction(filteredPods, rule, snap, action, backgroundCommandTermChan, rType, taskID)
+							err := executeCommandAction(filteredPods, rule, snap, action, backgroundPodListChan, rType, taskID)
 							if err != nil {
-								return backgroundCommandTermChan, err
+								if backgroundActionPresent {
```

Comment:
If any action fails we should terminate all the actions. If we are returning an error we shouldn't return any channels, its error prone since callers might just continue if err is set.

Reply:
done in the incremental